torch/csrc/jit/codegen/cuda/kernel_expr_evaluator.cpp

```diff
@@ -3,14 +3,49 @@
 #include <torch/csrc/jit/codegen/cuda/kernel_expr_evaluator.h>
 #include <torch/csrc/jit/codegen/cuda/kernel_ir_printer.h>
 
+#include <torch/csrc/jit/codegen/cuda/glfdc/eval.h>
+
 #include <iostream>
 
+namespace {
+
+using namespace torch::jit::fuser::cuda::kir;
+
+// KnownValuesCallback - Function object for retrieving binding runtime values
+class KnownValuesCallback {
+  using hashmap_t = std::unordered_map<const Val*, Int::ScalarType>;
+
+ public:
+  c10::optional<Int::ScalarType> operator()(uintptr_t binding) {
```

Comment:
Why is `uintptr_t` used? Why not just `const Val*`?

Reply:
At the cost of generality of using glfdc for different types, we could simply use kir::Val*.
The only assumption we have right now is that it is unique value for different symbolic values.

Reply:
Thanks. Please leave it as a comment somewhere.